File: leetcode/1523_count_odd_numbers_in_interval_range/solution.py
# Brute force solution that actually fails on leetcode becuase of timeout
def countOdds(low: int, high: int) -> int:
    count = 0

    for i in range(low, high + 1):
        if i % 2:
            count +=1

    return count

def countOdds2(low: int, high: int) -> int:
    # // is int division while / is returns a float
    ans = (high - low) // 2

    if low % 2 or high % 2:
        ans += 1

    return ans

assert(countOdds(3,7) == 3)
assert(countOdds2(3,7) == 3)

File: leetcode/1523_count_odd_numbers_in_interval_range/solution.go
package main

import "fmt"
import "time"

// Brute force solution
func countOdds(lo int, hi int) int {
    count := 0
    s := make([]int, hi-lo + 1)


    for i := range s {
        s[i] = lo + i
        if (lo + i) % 2 != 0 {
            count ++
        }
    }
    fmt.Println(count)
    return count
};

func main() {
    start := time.Now()
    // using 1, 700000000 as args takes this function almost 8 seconds to run
    countOdds(3, 7)
    fmt.Println(time.Since(start))
}
